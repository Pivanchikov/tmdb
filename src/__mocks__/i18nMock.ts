export default jest.mock('react-i18next', () => {
  return {
    useTranslation: () => {
      return {
        t: jest.fn(value => value),
        i18n: {
          language: 'en',
          changeLanguage: jest.fn(),
        },
      }
    },
  }
})
