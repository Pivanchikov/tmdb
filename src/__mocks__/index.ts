export * from './mockGenres'
export * from './mockMovies'
export * from './storeMock'
