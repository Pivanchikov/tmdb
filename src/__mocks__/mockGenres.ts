import { Genres, GenresSelectOption } from './types'

export const enGenresMock: Genres[] = [
  { id: 28, name: 'Action' },
  { id: 12, name: 'Adventure' },
  { id: 16, name: 'Animation' },
  { id: 35, name: 'Comedy' },
  { id: 80, name: 'Crime' },
  { id: 99, name: 'Documentary' },
  { id: 18, name: 'Drama' },
  { id: 10751, name: 'Family' },
  { id: 14, name: 'Fantasy' },
  { id: 36, name: 'History' },
  { id: 27, name: 'Horror' },
  { id: 10402, name: 'Music' },
  { id: 9648, name: 'Mystery' },
  { id: 10749, name: 'Romance' },
  { id: 878, name: 'Science Fiction' },
  { id: 10770, name: 'TV Movie' },
  { id: 53, name: 'Thriller' },
  { id: 10752, name: 'War' },
  { id: 37, name: 'Western' },
]

export const enGenresSelectMock: GenresSelectOption[] = [
  { value: 28, label: 'Action' },
  { value: 12, label: 'Adventure' },
  { value: 16, label: 'Animation' },
  { value: 35, label: 'Comedy' },
  { value: 80, label: 'Crime' },
  { value: 99, label: 'Documentary' },
  { value: 18, label: 'Drama' },
  { value: 10751, label: 'Family' },
  { value: 14, label: 'Fantasy' },
  { value: 36, label: 'History' },
  { value: 27, label: 'Horror' },
  { value: 10402, label: 'Music' },
  { value: 9648, label: 'Mystery' },
  { value: 10749, label: 'Romance' },
  { value: 878, label: 'Science Fiction' },
  { value: 10770, label: 'TV Movie' },
  { value: 53, label: 'Thriller' },
  { value: 10752, label: 'War' },
  { value: 37, label: 'Western' },
]
export const ruGenresMock: Genres[] = [
  { id: 28, name: 'боевик' },
  { id: 12, name: 'приключения' },
  { id: 16, name: 'мультфильм' },
  { id: 35, name: 'комедия' },
  { id: 80, name: 'криминал' },
  { id: 99, name: 'документальный' },
  { id: 18, name: 'драма' },
  { id: 10751, name: 'семейный' },
  { id: 14, name: 'фэнтези' },
  { id: 36, name: 'история' },
  { id: 27, name: 'ужасы' },
  { id: 10402, name: 'музыка' },
  { id: 9648, name: 'детектив' },
  { id: 10749, name: 'мелодрама' },
  { id: 878, name: 'фантастика' },
  { id: 10770, name: 'телевизионный фильм' },
  { id: 53, name: 'триллер' },
  { id: 10752, name: 'военный' },
  { id: 37, name: 'вестерн' },
]

export const emptyGenre: GenresSelectOption = { value: '', label: 'Any genre' }

export const ruGenresSelectMock: GenresSelectOption[] = [
  { value: 28, label: 'Боевик' },
  { value: 12, label: 'Приключения' },
  { value: 16, label: 'Мультфильм' },
  { value: 35, label: 'Комедия' },
  { value: 80, label: 'Криминал' },
  { value: 99, label: 'Документальный' },
  { value: 18, label: 'Драма' },
  { value: 10751, label: 'Семейный' },
  { value: 14, label: 'Фэнтези' },
  { value: 36, label: 'История' },
  { value: 27, label: 'Ужасы' },
  { value: 10402, label: 'Музыка' },
  { value: 9648, label: 'Детектив' },
  { value: 10749, label: 'Мелодрама' },
  { value: 878, label: 'Фантастика' },
  { value: 10770, label: 'Телевизионный фильм' },
  { value: 53, label: 'Триллер' },
  { value: 10752, label: 'Военный' },
  { value: 37, label: 'Вестерн' },
]
