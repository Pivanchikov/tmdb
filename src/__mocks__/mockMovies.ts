/* eslint-disable no-magic-numbers */
import { TransformMovies } from './types'
import { Movie } from '../state/types'

export const movieWithAllParam: Movie = {
  adult: false,
  backdrop_path: '/las0P4Dua54XrZ73VQmGUaH1z0U.jpg',
  genre_ids: [16, 28, 18, 878],
  id: 283566,
  original_language: 'ja',
  original_title: 'シン・エヴァンゲリオン劇場版:||',
  overview:
    'In the aftermath of the Fourth Impact, stranded without their Evangelions, Shinji, Asuka, and Rei search for refuge in the desolate red remains of Tokyo-3. But the danger to the world is far from over. A new impact is looming on the horizon—one that will prove to be the true end of Evangelion.',
  popularity: 1125.051,
  poster_path: '/jDwZavHo99JtGsCyRzp4epeeBHx.jpg',
  release_date: '2021-03-08',
  title: 'Evangelion: 3.0+1.0 Thrice Upon a Time',
  video: false,
  vote_average: 8.8,
  vote_count: 194,
}

export const movieWithOutAllParam: Movie = {
  adult: false,
  backdrop_path: '/las0P4Dua54XrZ73VQmGUaH1z0U.jpg',
  genre_ids: [16, 28, 18, 878],
  id: 283566,
  original_language: 'ja',
  original_title: 'シン・エヴァンゲリオン劇場版:||',
  overview: null,
  popularity: 1125.051,
  poster_path: null,
  release_date: '2021-03-08',
  title: 'Evangelion: 3.0+1.0 Thrice Upon a Time',
  video: false,
  vote_average: 8,
  vote_count: 194,
}

export const movies: Movie[] = [
  {
    adult: false,
    backdrop_path: '/las0P4Dua54XrZ73VQmGUaH1z0U.jpg',
    genre_ids: [16, 28, 18, 878],
    id: 283566,
    original_language: 'ja',
    original_title: 'シン・エヴァンゲリオン劇場版:||',
    overview:
      'In the aftermath of the Fourth Impact, stranded without their Evangelions, Shinji, Asuka, and Rei search for refuge in the desolate red remains of Tokyo-3. But the danger to the world is far from over. A new impact is looming on the horizon—one that will prove to be the true end of Evangelion.',
    popularity: 1125.051,
    poster_path: '/jDwZavHo99JtGsCyRzp4epeeBHx.jpg',
    release_date: '2021-03-08',
    title: 'Evangelion: 3.0+1.0 Thrice Upon a Time',
    video: false,
    vote_average: 8.8,
    vote_count: 194,
  },
  {
    adult: false,
    backdrop_path: '/6MKr3KgOLmzOP6MSuZERO41Lpkt.jpg',
    genre_ids: [35, 80],
    id: 337404,
    original_language: 'en',
    original_title: 'Cruella',
    overview:
      'In 1970s London amidst the punk rock revolution, a young grifter named Estella is determined to make a name for herself with her designs. She befriends a pair of young thieves who appreciate her appetite for mischief, and together they are able to build a life for themselves on the London streets. One day, Estella’s flair for fashion catches the eye of the Baroness von Hellman, a fashion legend who is devastatingly chic and terrifyingly haute. But their relationship sets in motion a course of events and revelations that will cause Estella to embrace her wicked side and become the raucous, fashionable and revenge-bent Cruella.',
    popularity: 978.832,
    poster_path: '/wToO8opxkGwKgSfJ1JK8tGvkG6U.jpg',
    release_date: '2021-05-26',
    title: 'Cruella',
    video: false,
    vote_average: 8.4,
    vote_count: 4777,
  },
  {
    adult: false,
    backdrop_path: '/8s4h9friP6Ci3adRGahHARVd76E.jpg',
    genre_ids: [16, 35, 10751, 878],
    id: 379686,
    original_language: 'en',
    original_title: 'Space Jam: A New Legacy',
    overview:
      "When LeBron and his young son Dom are trapped in a digital space by a rogue A.I., LeBron must get them home safe by leading Bugs, Lola Bunny and the whole gang of notoriously undisciplined Looney Tunes to victory over the A.I.'s digitized champions on the court. It's Tunes versus Goons in the highest-stakes challenge of his life.",
    popularity: 2105.173,
    poster_path: '/5bFK5d3mVTAvBCXi5NPWH0tYjKl.jpg',
    release_date: '2021-07-08',
    title: 'Space Jam: A New Legacy',
    video: false,
    vote_average: 7.5,
    vote_count: 1774,
  },
  {
    adult: false,
    backdrop_path: '/xXHZeb1yhJvnSHPzZDqee0zfMb6.jpg',
    genre_ids: [28, 80, 53],
    id: 385128,
    original_language: 'en',
    original_title: 'F9',
    overview:
      "Dominic Toretto and his crew battle the most skilled assassin and high-performance driver they've ever encountered: his forsaken brother.",
    popularity: 2119.117,
    poster_path: '/bOFaAXmWWXC3Rbv4u4uM9ZSzRXP.jpg',
    release_date: '2021-05-19',
    title: 'F9',
    video: false,
    vote_average: 7.7,
    vote_count: 2804,
  },
  {
    adult: false,
    backdrop_path: '/jlGmlFOcfo8n5tURmhC7YVd4Iyy.jpg',
    genre_ids: [28, 12, 14],
    id: 436969,
    original_language: 'en',
    original_title: 'The Suicide Squad',
    overview:
      'Supervillains Harley Quinn, Bloodsport, Peacemaker and a collection of nutty cons at Belle Reve prison join the super-secret, super-shady Task Force X as they are dropped off at the remote, enemy-infused island of Corto Maltese.',
    popularity: 6294.822,
    poster_path: '/iCi4c4FvVdbaU1t8poH1gvzT6xM.jpg',
    release_date: '2021-07-28',
    title: 'The Suicide Squad',
    video: false,
    vote_average: 8.1,
    vote_count: 2435,
  },
]

export const transformMovies: TransformMovies = {
  '283566': {
    adult: false,
    backdrop_path: '/las0P4Dua54XrZ73VQmGUaH1z0U.jpg',
    genre_ids: [16, 28, 18, 878],
    id: 283566,
    original_language: 'ja',
    original_title: 'シン・エヴァンゲリオン劇場版:||',
    overview:
      'In the aftermath of the Fourth Impact, stranded without their Evangelions, Shinji, Asuka, and Rei search for refuge in the desolate red remains of Tokyo-3. But the danger to the world is far from over. A new impact is looming on the horizon—one that will prove to be the true end of Evangelion.',
    popularity: 1125.051,
    poster_path: '/jDwZavHo99JtGsCyRzp4epeeBHx.jpg',
    release_date: '2021-03-08',
    title: 'Evangelion: 3.0+1.0 Thrice Upon a Time',
    video: false,
    vote_average: 8.8,
    vote_count: 194,
  },
  '337404': {
    adult: false,
    backdrop_path: '/6MKr3KgOLmzOP6MSuZERO41Lpkt.jpg',
    genre_ids: [35, 80],
    id: 337404,
    original_language: 'en',
    original_title: 'Cruella',
    overview:
      'In 1970s London amidst the punk rock revolution, a young grifter named Estella is determined to make a name for herself with her designs. She befriends a pair of young thieves who appreciate her appetite for mischief, and together they are able to build a life for themselves on the London streets. One day, Estella’s flair for fashion catches the eye of the Baroness von Hellman, a fashion legend who is devastatingly chic and terrifyingly haute. But their relationship sets in motion a course of events and revelations that will cause Estella to embrace her wicked side and become the raucous, fashionable and revenge-bent Cruella.',
    popularity: 978.832,
    poster_path: '/wToO8opxkGwKgSfJ1JK8tGvkG6U.jpg',
    release_date: '2021-05-26',
    title: 'Cruella',
    video: false,
    vote_average: 8.4,
    vote_count: 4777,
  },
  '379686': {
    adult: false,
    backdrop_path: '/8s4h9friP6Ci3adRGahHARVd76E.jpg',
    genre_ids: [16, 35, 10751, 878],
    id: 379686,
    original_language: 'en',
    original_title: 'Space Jam: A New Legacy',
    overview:
      "When LeBron and his young son Dom are trapped in a digital space by a rogue A.I., LeBron must get them home safe by leading Bugs, Lola Bunny and the whole gang of notoriously undisciplined Looney Tunes to victory over the A.I.'s digitized champions on the court. It's Tunes versus Goons in the highest-stakes challenge of his life.",
    popularity: 2105.173,
    poster_path: '/5bFK5d3mVTAvBCXi5NPWH0tYjKl.jpg',
    release_date: '2021-07-08',
    title: 'Space Jam: A New Legacy',
    video: false,
    vote_average: 7.5,
    vote_count: 1774,
  },
  '385128': {
    adult: false,
    backdrop_path: '/xXHZeb1yhJvnSHPzZDqee0zfMb6.jpg',
    genre_ids: [28, 80, 53],
    id: 385128,
    original_language: 'en',
    original_title: 'F9',
    overview:
      "Dominic Toretto and his crew battle the most skilled assassin and high-performance driver they've ever encountered: his forsaken brother.",
    popularity: 2119.117,
    poster_path: '/bOFaAXmWWXC3Rbv4u4uM9ZSzRXP.jpg',
    release_date: '2021-05-19',
    title: 'F9',
    video: false,
    vote_average: 7.7,
    vote_count: 2804,
  },
  '436969': {
    adult: false,
    backdrop_path: '/jlGmlFOcfo8n5tURmhC7YVd4Iyy.jpg',
    genre_ids: [28, 12, 14],
    id: 436969,
    original_language: 'en',
    original_title: 'The Suicide Squad',
    overview:
      'Supervillains Harley Quinn, Bloodsport, Peacemaker and a collection of nutty cons at Belle Reve prison join the super-secret, super-shady Task Force X as they are dropped off at the remote, enemy-infused island of Corto Maltese.',
    popularity: 6294.822,
    poster_path: '/iCi4c4FvVdbaU1t8poH1gvzT6xM.jpg',
    release_date: '2021-07-28',
    title: 'The Suicide Squad',
    video: false,
    vote_average: 8.1,
    vote_count: 2435,
  },
}

export const mockMovies = [[movies, transformMovies]]
