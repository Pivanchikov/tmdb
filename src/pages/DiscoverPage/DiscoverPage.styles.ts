export const styles = theme => ({
  container: {
    minHeight: 'calc(100vh - 128px)',
  },

  linksContainer: {
    display: 'flex',
    width: 370,
    margin: '0 auto',
  },

  button: {
    width: 150,
    backgroundColor: theme.palette.primary.main,
    display: 'block',
    color: theme.palette.secondary.main,
    margin: '20px auto',
    '&:hover': {
      backgroundColor: theme.palette.primary.light,
    },
  },

  center: {
    paddingTop: '1.5rem',
    margin: '0 auto',
    width: 740,
  },

  year: {
    margin: '0 auto',
    width: 740,
  },

  loader: {
    display: 'flex',
    justifyContent: 'center',
    margin: '1rem',
  },

  '@media screen and (max-width: 745px)': {
    container: {
      minHeight: 'calc(100vh - 176px)',
    },
  },

  '@media screen and (max-width: 959px)': {
    center: {
      display: 'flex',
      flexWrap: 'wrap',
      width: 'auto',
      justifyContent: 'center',
    },
    year: {
      display: 'flex',
      flexWrap: 'wrap',
      width: 'auto',
      justifyContent: 'center',
    },
  },
})
