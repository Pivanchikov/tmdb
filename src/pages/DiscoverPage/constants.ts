export const STEP = 1
export const MIN_YEAR = 1920
export const MAX_YEAR = 2021
