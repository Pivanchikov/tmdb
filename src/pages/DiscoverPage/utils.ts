import { MIN_YEAR, MAX_YEAR } from './constants'

interface RangePickerOption {
  value: number
  label: string
}

export const getYearsOptions = (start: number = MIN_YEAR, end: number = MAX_YEAR): RangePickerOption[] =>
  // eslint-disable-next-line no-magic-numbers
  new Array(end - start).fill(0).map((_, index) => {
    const year = start + index

    return { value: year, label: `${year}` }
  })
