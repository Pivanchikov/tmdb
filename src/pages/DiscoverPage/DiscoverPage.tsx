import { useState, useEffect, useMemo, useCallback } from 'react'
import { Button, makeStyles, Hidden, CircularProgress } from '@material-ui/core'
import { useSelector, useDispatch } from 'react-redux'
import { useTranslation } from 'react-i18next'

import { getGenresItems, fetchGenres, fetchSingleMovie, getIsLoading, getIsFoundMovie } from '../../state'
import { RangePicker, Dropdown } from '../../components'
import { FoundMovie } from './FoundMovie/FoundMovie'
import { styles } from './DiscoverPage.styles'
import { MIN_YEAR, MAX_YEAR, STEP } from './constants'
import { getYearsOptions } from './utils'

const useStyles = makeStyles(styles)

export const DiscoverPage = () => {
  const isFoundMovie = useSelector(getIsFoundMovie)

  const { t } = useTranslation()
  const genres = useSelector(getGenresItems)
  const isLoading = useSelector(getIsLoading)
  const countries = useMemo(
    () => [
      { value: 'AT', label: t('austria') },
      { value: 'BE', label: t('belgium') },
      { value: 'CA', label: t('сanada') },
      { value: 'CN', label: t('сhina') },
      { value: 'DK', label: t('denmark') },
      { value: 'FR', label: t('france') },
      { value: 'DE', label: t('germany') },
      { value: 'HK', label: t('hongKong') },
      { value: 'IE', label: t('ireland') },
      { value: 'IT', label: t('italy') },
      { value: 'JP', label: t('japan') },
      { value: 'KR', label: t('koreaSouth') },
      { value: 'MX', label: t('mexico') },
      { value: 'RU', label: t('russia') },
      { value: 'ES', label: t('spain') },
      { value: 'SE', label: t('sweden') },
      { value: 'GB', label: t('unitedKingdom') },
      { value: 'US', label: t('usa') },
    ],
    [t]
  )

  const years = useMemo(getYearsOptions, [])
  const classes = useStyles()
  const dispatch = useDispatch()
  const [genre, setGenre] = useState('')
  const [country, setCountry] = useState('')
  const [year, setYear] = useState('')

  const label = `${t('productionYear')}: ${year || t('any')}`

  const handleGenreChange = useCallback(genre => setGenre(genre.value), [])
  const handleCountryChange = useCallback(country => setCountry(country.value), [])
  const handleYearsChange = useCallback(years => {
    if (Array.isArray(years)) {
      const [value] = years
      setYear(value)
    } else {
      setYear(years.value)
    }
  }, [])

  const search = () => {
    dispatch(fetchSingleMovie({ genre, country, year }))
  }

  const rangePickerValue = year ? [+year] : [MIN_YEAR]

  useEffect(() => {
    dispatch(fetchGenres())
  }, [dispatch, t])

  return (
    <section className={classes.container}>
      <div className={classes.center}>
        <Dropdown
          options={genres}
          onChange={handleGenreChange}
          emptyValue={{ value: '', label: `${t('anyGenre')}` }}
          testId="selectGenres"
        />
        <Dropdown
          options={countries}
          onChange={handleCountryChange}
          emptyValue={{ value: '', label: `${t('anyCountry')}` }}
          testId="selectCountry"
        />
      </div>
      <div className={classes.year}>
        <Hidden smDown>
          <RangePicker
            step={STEP}
            min={MIN_YEAR}
            max={MAX_YEAR}
            values={rangePickerValue}
            onChange={handleYearsChange}
            label={label}
          />
        </Hidden>
        <Hidden mdUp>
          <Dropdown
            options={years}
            defaultValue={year}
            onChange={handleYearsChange}
            emptyValue={{ value: '', label: `${t('anyYear')}` }}
            testId="selectYear"
          />
        </Hidden>
      </div>
      <Button onClick={search} className={classes.button} data-testid={'buttonFind'}>
        {t('find')}
      </Button>
      {isLoading && (
        <div className={classes.loader}>
          <CircularProgress color="primary" className={classes.loader} />
        </div>
      )}
      {isFoundMovie && (
        <>
          <FoundMovie />
          <div className={classes.linksContainer}>
            <Button className={classes.button} data-testid={'popularLink'}>
              {t('popular')}
            </Button>
            <Button className={classes.button} data-testid={'topRatedLink'}>
              {t('topRated')}
            </Button>
          </div>
        </>
      )}
    </section>
  )
}
