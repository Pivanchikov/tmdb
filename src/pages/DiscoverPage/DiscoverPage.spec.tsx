import { render, fireEvent } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import { DiscoverPage } from './DiscoverPage'
import { initialState } from '../../__mocks__'

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

type DropdownProps = {
  onChange: (option: { value: string; label: string } | undefined) => void
  testId: string
  options: { value: string; label: string }[]
  value: string | number | readonly string[]
}

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: jest.fn(value => value),
    i18n: {
      language: 'en',
      changeLanguage: jest.fn(),
    },
  }),
}))

jest.mock('../../components', () => ({
  Dropdown({ options, value, onChange, testId }: DropdownProps) {
    function handleChange(event) {
      const option = options.find(option => {
        return option.value == event.currentTarget.value
      })

      onChange(option)
    }

    return (
      <select data-testid={testId} value={value} onChange={event => handleChange(event)}>
        {options?.map(({ label, value }) => (
          <option key={value} value={value}>
            {label}
          </option>
        ))}
      </select>
    )
  },
  RangePicker() {
    return <div>RangePicker</div>
  },
}))

describe('Search page', () => {
  const store = mockStore({
    movies: initialState,
  })

  const setup = () =>
    render(
      <Provider store={store}>
        <DiscoverPage />
      </Provider>
    )

  it('render foundMovie component', () => {
    const { getByTestId } = setup()

    fireEvent.click(getByTestId('buttonFind'))
  })
  it('change value for select', () => {
    const { getByTestId } = setup()

    fireEvent.change(getByTestId('selectCountry'), { value: '', label: 'test' })
  })
  it('matches snapshot', () => {
    const { asFragment } = setup()

    expect(asFragment()).toMatchSnapshot()
  })
})
