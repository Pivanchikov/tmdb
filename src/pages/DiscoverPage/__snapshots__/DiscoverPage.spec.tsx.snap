// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search page matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="makeStyles-container-13"
  >
    <div
      class="makeStyles-center-16"
    >
      <select
        data-testid="selectGenres"
      />
      <select
        data-testid="selectCountry"
      >
        <option
          value="AT"
        >
          austria
        </option>
        <option
          value="BE"
        >
          belgium
        </option>
        <option
          value="CA"
        >
          сanada
        </option>
        <option
          value="CN"
        >
          сhina
        </option>
        <option
          value="DK"
        >
          denmark
        </option>
        <option
          value="FR"
        >
          france
        </option>
        <option
          value="DE"
        >
          germany
        </option>
        <option
          value="HK"
        >
          hongKong
        </option>
        <option
          value="IE"
        >
          ireland
        </option>
        <option
          value="IT"
        >
          italy
        </option>
        <option
          value="JP"
        >
          japan
        </option>
        <option
          value="KR"
        >
          koreaSouth
        </option>
        <option
          value="MX"
        >
          mexico
        </option>
        <option
          value="RU"
        >
          russia
        </option>
        <option
          value="ES"
        >
          spain
        </option>
        <option
          value="SE"
        >
          sweden
        </option>
        <option
          value="GB"
        >
          unitedKingdom
        </option>
        <option
          value="US"
        >
          usa
        </option>
      </select>
    </div>
    <div
      class="makeStyles-year-17"
    />
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-15"
      data-testid="buttonFind"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        find
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </section>
</DocumentFragment>
`;
